fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/VanillaToolsSupport.java

```diff
@@ -0,0 +1,63 @@
+package net.fabricmc.fabric.impl.tool.attribute;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.Material;
+import net.minecraft.item.Item;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.event.registry.RegistryEntryAddedCallback;
+import net.fabricmc.fabric.api.tool.attribute.v1.FabricToolTags;
+
+public class VanillaToolsSupport implements ModInitializer {
+	private static final Logger LOGGER = LogManager.getLogger("fabric-tool-attribute-api-v1");
+
+	@Override
+	public void onInitialize() {
+		for (Block block : Registry.BLOCK) {
+			registerBlockSupport(Registry.BLOCK.getId(block), block);
+		}
+
+		RegistryEntryAddedCallback.event(Registry.BLOCK).register((rawId, id, object) -> registerBlockSupport(id, object));
```

Comment:
I'm not sure where in loading stages ModInitializer gets called - is it before vanilla reg, after it, or unreliable?

Reply:
Mod initializers are run after vanilla registries have been bootstrapped.

Reply:
Ah right, I will remove this